vpn_slice/__main__.py

```diff
@@ -138,6 +138,15 @@ def do_disconnect(env, args):
     except CalledProcessError:
         print("WARNING: could not delete route to VPN gateway (%s)" % env.gateway, file=stderr)
 
+    for dest in args.exc_subnets:
+        providers.route.remove_route(dest)
+
+    for dest in args.subnets:
+        providers.route.remove_route(dest)
```

Comment:
Some of these are routes that have explicitly been set to go through the outer/Internet-facing network interface, and some have explicitly been set to go through the inner/VPN-tunnel interface.

The `remove_route` method should probably be augmented with `via`, `dev` parameters, like `add_route`/`replace_route` have:
https://github.com/dlenski/vpn-slice/blob/ba76f82d8a71514db08dbfba1b78cb00a986827d/vpn_slice/linux.py#L52-L59

And the appropriate parameters should be specified with `remove_route`. Otherwise, we run the risk of deleting routes on the wrong interface.

Reply:
Also, `remove_route` can fail (e.g. if the route has already been deleted, which may happen depending on the exact timing of tunnel interface teardown on some OSes), so let's please wrap these in `try: … except: print("WARNING: ...")` as for the gateway route case.